aws/resource_aws_lb.go

```diff
@@ -74,14 +74,11 @@ func resourceAwsLb() *schema.Resource {
 			},
 
 			"load_balancer_type": {
-				Type:     schema.TypeString,
-				ForceNew: true,
-				Optional: true,
-				Default:  elbv2.LoadBalancerTypeEnumApplication,
-				ValidateFunc: validation.StringInSlice([]string{
-					elbv2.LoadBalancerTypeEnumApplication,
-					elbv2.LoadBalancerTypeEnumNetwork,
-				}, false),
+				Type:         schema.TypeString,
+				ForceNew:     true,
+				Optional:     true,
+				Default:      elbv2.LoadBalancerTypeEnumApplication,
+				ValidateFunc: validation.StringInSlice(elbv2.LoadBalancerTypeEnum_Values(), false),
```

Comment:
I like the use of `_Values()` generally but here where it affects so much of the function of the resource and a new value will almost certainly break things, it's worth considering a hardcoded slice.

Reply:
Do you have a specific case in mind? I didn't see any regressions from these additions and eventually this will be reported by https://github.com/hashicorp/terraform-provider-aws/issues/14601. If new automatically added types do not work correctly without additional rework, then its no different than our manual feature request efforts today.

Reply:
My suggestion in this case is if you see logic that is not being defensive about new values being either manually or automatically introduced (in this or any other resource), to create technical debt issues. 👍 

In overall scalability terms, we really want to start leaning on automatic code updates where possible even if it changes the dynamic of whether issues get reported as strictly feature requests or potentially as "bugs" when new functionality is partially enabled.